internal/translator/data/rules_test.go

```diff
@@ -0,0 +1,232 @@
+	helper := Rules{
+		Collection: collection,
+	}
+
+	defer func(t *testing.T) {
+		t.Logf("Cleaning the database")
+		helper.Collection.Drop(context.Background())
+		testInfo.db.Disconnect(context.Background())
+	}(t)
+
+	t.Run("save a new rule with success when the strucutre is valid", func(t *testing.T) {
+		newRule := Rule{
+			SourceLanguage: "espaol",
+			TargetLanguage: "kaqchikel",
+			Pattern:        "VERB,ADV,ADJ",
+			Details: []RuleDetail{
+				{
+					Tag:  "VERB",
+					Type: "A",
+				},
+				{
+					Tag:  "ADV",
+					Type: "G",
+				},
+				{
+					Tag:  "ADJ",
+					Type: "Q",
+				},
+			},
+			Output: []RuleOutput{
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 2).Lemma}}",
+				},
+				{
+					"type":  "literal",
+					"value": " ",
+				},
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 1).Lemma}}",
+				},
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 0).Lemma}}",
+				},
+			},
+		}
+
+		newID, error := helper.NewRule(context.Background(), newRule)
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("New rule created with id %s", newID.Hex())
+	})
+
+	t.Run("save a new rule with success and with the same last pattern but different detail when the strucutre is valid", func(t *testing.T) {
+		newRule := Rule{
+			SourceLanguage: "espaol",
+			TargetLanguage: "kaqchikel",
+			Pattern:        "VERB,ADV,ADJ",
+			Details: []RuleDetail{
+				{
+					Tag:  "VERB",
+					Type: "M",
+				},
+				{
+					Tag:  "ADV",
+					Type: "G",
+				},
+				{
+					Tag:  "ADJ",
+					Type: "O",
+				},
+			},
+			Output: []RuleOutput{
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 1).Lemma}}",
+				},
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 0).Lemma}}",
+				},
+			},
+		}
+
+		newID, error := helper.NewRule(context.Background(), newRule)
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("New rule created with id %s", newID.Hex())
+	})
+
+	t.Run("find rule with success when the value exists", func(t *testing.T) {
+		rule, error := helper.FindRuleByPattern(context.Background(), "espaol", "kaqchikel", "VERB,ADV,ADJ")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("Found rules %v", rule)
+	})
+
+	t.Run("get and error finding non existing rule", func(t *testing.T) {
+		rule, error := helper.FindRuleByPattern(context.Background(), "espaol", "kaqchikel", "ADJ,ADJ,ADV")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		if len(rule) == 0 {
+			t.Logf("pattern ADJ,ADJ,ADV doesn't exist")
+		}
+	})
+
+	t.Run("delete a group of rules with success when the rules exist", func(t *testing.T) {
+		error := helper.DeleteMany(context.Background(), "espaol", "kaqchikel", "VERB,ADV,ADJ")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("Rules deleted")
+	})
+
+	t.Run("add short new rule with success when the strucutre is valid", func(t *testing.T) {
```

Comment:
Some as above, this test the same feature as the first one. This should be deleted.

Reply:
I need more than one rule for testing with more data, not only one, that's the reason a new rule and shorter was created, I think is good

Reply:
Go it.

If you want you can leave it, it's not a critical change.

But my point is, what is the functional difference? Basically both are valid because you are testing the same functionality due we are not validating types or lengths. If you save an array of 0 or 100 elements within it for the function it's the same.